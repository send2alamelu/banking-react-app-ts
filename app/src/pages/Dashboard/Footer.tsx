import { useHistory } from 'react-router-dom';

import BottomNavigationAction from '@mui/material/BottomNavigationAction';
import SendIcon from '@mui/icons-material/Send';

import BottomNav from '../../components/Common/BottomNav/BottomNav';

import { ROUTE_PATH } from '../../constants/RoutePath';
const { TRANSACTION } = ROUTE_PATH;

export default function Footer() {
  const history = useHistory();

  const goToTransferForm = (e: React.MouseEvent<HTMLElement>) => {
    e.preventDefault();
    history.push(TRANSACTION);
  }

  return (
    <BottomNav>
      <BottomNavigationAction label="Make a transfer" onClick={goToTransferForm} icon={<SendIcon />} />
    </BottomNav>
  );
}
