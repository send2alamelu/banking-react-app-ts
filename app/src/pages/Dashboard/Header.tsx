import React from 'react';
import { useHistory } from 'react-router-dom';
import Link from '@mui/material/Link';

import { ROUTE_PATH } from '../../constants/RoutePath';
const { LOGIN } = ROUTE_PATH;

export default function Header() {
  const history = useHistory();
  const logout = (e: any) => {
    e.preventDefault();
    localStorage.removeItem('jwtToken');
    history.push(LOGIN);
  }

  return (
    <Link
      component="button"
      variant="body2"
      onClick={logout}
    >
      Logout
    </Link>
  );
}