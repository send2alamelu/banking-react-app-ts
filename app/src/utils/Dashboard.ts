import format from 'date-fns/format'

export const formatTransactionDate = (transactionDate: string) => {
    return format(new Date(transactionDate), 'd MMM')
}