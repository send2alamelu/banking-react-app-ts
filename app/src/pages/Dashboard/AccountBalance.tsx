import { useState, useEffect } from 'react';
import Typography from '@mui/material/Typography';

import useGet from '../../hooks/useGet';
import { getBalancesApiPath } from '../../utils/Api';
import { currencyFormatter } from '../../utils/Transaction';
import { GetResponse } from '../../types/ApiResponse';

import Block from '../../components/Common/Block/Block';

export default function AccountBalance() {
  const getBalance: GetResponse = useGet(getBalancesApiPath());
  const balanceResponse: any = getBalance?.response || {}
  const [balanceAmount, setBalanceAmount] = useState('');

  useEffect(() => {
    const { balance } = balanceResponse;
    if (balance) {
      setBalanceAmount(
        currencyFormatter(balance)
      )
    }
  }, [balanceResponse])


  return (
    <Block>
      <Typography sx={{ fontSize: 14 }} color="text.secondary" gutterBottom>
        You have
      </Typography>
      <Typography variant="h5" component="div">
        {balanceAmount}
      </Typography>
      <Typography variant="body2">
        in your account
      </Typography>
    </Block>
  );
}
