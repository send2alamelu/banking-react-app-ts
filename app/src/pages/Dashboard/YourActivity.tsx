import { useEffect, useState } from 'react';
import Divider from '@mui/material/Divider';
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableRow from '@mui/material/TableRow';
import Paper from '@mui/material/Paper';

import useGet from '../../hooks/useGet';
import { getTransactionApiPath } from '../../utils/Api';
import { GetResponse } from '../../types/ApiResponse';
import { TransactionItem } from '../../types/TransactionItem';
import { getDescriptionFromTransactionType } from '../../utils/Transaction';

export default function YourActivity() {
  const getTransaction: GetResponse = useGet(getTransactionApiPath());
  const [transactionList, setTransactionList] = useState([]);

  useEffect(() => {
    const transactions = getTransaction?.response?.data || []
    if (transactions.length > 0) {
      setTransactionList(transactions);
    }

  }, [getTransaction?.response]);

  return (
    <>
      <h1>Your Activity</h1>
      <Divider />
      <TableContainer component={Paper}>
        <Table aria-label="activity table">
          <TableBody>
            {transactionList.map((transaction: TransactionItem) => (
              <TableRow
                key={transaction.amount}
                sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
              >
                <TableCell component="th" scope="row">
                  {transaction.date}
                </TableCell>
                <TableCell align="left">{getDescriptionFromTransactionType(transaction)}</TableCell>
                <TableCell align="left">{transaction.amount}</TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </TableContainer>
    </>
  );
}
