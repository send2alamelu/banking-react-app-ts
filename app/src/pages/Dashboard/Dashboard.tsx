import React from 'react';

import Block from '../../Components/Common/Block/Block'
import YourActivity from './YourActivity'
import Header from './Header'
import Footer from './Footer'

function Dashboard() {
  return (
    <div className="Dashboard">
      <Header />
      <Block />
      <YourActivity />
      <Footer />
    </div>
  );
}

export default Dashboard;
