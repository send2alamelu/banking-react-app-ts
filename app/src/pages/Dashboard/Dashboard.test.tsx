import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Dashboard from './Dashboard';

describe('Dashboard Component', () => {
  afterEach(() => {
    cleanup();
  });

  test('renders Dashboard with children', () => {
    const { asFragment } = render(<Dashboard />);
    expect(asFragment()).toMatchSnapshot();
  });
});