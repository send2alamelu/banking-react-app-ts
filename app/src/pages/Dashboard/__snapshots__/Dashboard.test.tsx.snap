// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Component renders Dashboard with children 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined sc-gsDKAQ fNHwLb css-11lq3yg-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 sc-gsDKAQ fNHwLb css-pavvoq-MuiGrid-root"
    >
      <div
        class="sc-bdvvtL eLWaJE MuiBox-root css-0"
      >
        <button
          class="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways MuiLink-button css-1daramq-MuiTypography-root-MuiLink-root"
        >
          Logout
        </button>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 sc-gsDKAQ fNHwLb css-16nn1fs-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-16m3jxf-MuiTypography-root"
          >
            You have
          </p>
          <div
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          />
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            in your account
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 sc-gsDKAQ fNHwLb css-16nn1fs-MuiGrid-root"
    >
      <h1>
        Your Activity
      </h1>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
      >
        <table
          aria-label="activity table"
          class="MuiTable-root css-rqglhn-MuiTable-root"
        >
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          />
        </table>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 sc-gsDKAQ fNHwLb css-16nn1fs-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-xuhmcu"
      >
        <div
          class="MuiBottomNavigation-root css-16lloyr-MuiBottomNavigation-root"
        >
          <button
            class="MuiButtonBase-root MuiBottomNavigationAction-root css-1ee5err-MuiButtonBase-root-MuiBottomNavigationAction-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SendIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
              />
            </svg>
            <span
              class="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
            >
              Make a transfer
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
